src/org/opendatakit/briefcase/reused/http/Credentials.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2018 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.opendatakit.briefcase.reused.http;
+
+import java.util.Objects;
+
+/**
+ * This Value Object class holds a username/password pair to be used as credentials.
+ */
+public class Credentials {
+  private final String username;
+  private final String password;
+
+  public Credentials(String username, String password) {
+    this.username = username;
+    this.password = password;
+  }
+
+  public String getUsername() {
+    return username;
+  }
+
+  public String getPassword() {
+    return password;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    Credentials that = (Credentials) o;
+    return Objects.equals(username, that.username) &&
+        Objects.equals(password, that.password);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(username, password);
```

Comment:
I see you’re using `Objects.hash`. Would `Objects.deepEquals` be useful, too?

Reply:
I don't know about that. I override `.equals` and `.hash` to make Value Objects. Any member of a Value Object must be a primitive or another Value Object for this to work, therefore, `.equals` would and should be enough.